pkg/util/azureclient/compute_addons.go

```diff
@@ -71,6 +78,66 @@ func (c *virtualMachineScaleSetsClient) UpdateInstances(ctx context.Context, res
 	return future.WaitForCompletionRef(ctx, c.VirtualMachineScaleSetsClient.Client)
 }
 
+type ListScaleSetPublicIPsFuture struct {
+	azure.Future
+}
+
+// ListScaleSetPublicIPs lists all public IP addresses used by VMSS.
+// Current golang Azure SDK does not have ListScaleSetPublicIPs as a method.
+// https://github.com/Azure/azure-sdk-for-go/issues/4305
+func (c *virtualMachineScaleSetsClient) ListScaleSetPublicIPs(ctx context.Context, resourceGroupName, VMScaleSetName string) ([]string, error) {
+	req, err := c.ListScaleSetPublicIPsPreparer(ctx, resourceGroupName, VMScaleSetName)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.ListScaleSetPublicIPsSender(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var body interface{}
+	defer response.Body.Close()
+	err = json.NewDecoder(response.Body).Decode(&body)
+	if err != nil {
+		return nil, err
+	}
+
+	var ips []string
+	for _, ip := range jsonpath.MustCompile("$.value[*].properties.ipAddress").Get(body) {
```

Comment:
this is a hack, and the bar to entry to this file is higher.  I'd just keep all this out of azureclient as it was in the file I sent you